sqlmesh/core/engine_adapter/base.py

```diff
@@ -1208,23 +1267,131 @@ def scd_type_2(
+            )
+        if execution_time_as_valid_from and not check_columns:
+            raise SQLMeshError(
+                "Cannot use `execution_time_as_valid_from` without `check_columns` for SCD Type 2"
+            )
+        if updated_at_name and updated_at_name not in columns_to_types:
             raise SQLMeshError(
                 f"Column {updated_at_name} not found in {target_table}. Table must contain an `updated_at` timestamp for SCD Type 2"
             )
+
         unmanaged_columns = [
             col for col in columns_to_types if col not in {valid_from_name, valid_to_name}
         ]
         time_data_type = columns_to_types[valid_from_name]
         select_source_columns: t.List[t.Union[str, exp.Alias]] = [
             col for col in unmanaged_columns if col != updated_at_name
         ]
-        select_source_columns.append(exp.cast(updated_at_name, time_data_type).as_(updated_at_name))
-        valid_from_start = (
-            updated_at_name
-            if updated_at_as_valid_from
-            else self._to_utc_timestamp("1970-01-01 00:00:00+00:00", time_data_type)
-        )
+        if updated_at_name:
+            select_source_columns.append(
+                exp.cast(updated_at_name, time_data_type).as_(updated_at_name)
+            )
+
+        # If a star is provided, we include all unmanaged columns in the check.
+        # This unnecessarily includes unique key columns but since they are used in the join, and therefore we know
+        # they are equal or not, the extra check is not a problem and we gain simplified logic here.
+        # If we want to change this, then we just need to check the expressions in unique_key and pull out the
+        # column names and then remove them from the unmanaged_columns
+        if check_columns and check_columns == exp.Star():
+            check_columns = [exp.column(col) for col in unmanaged_columns]
+        execution_ts = self._to_utc_timestamp(to_ts(execution_time), time_data_type)
+        update_valid_from_start: t.Union[str, exp.Expression]
+        if updated_at_as_valid_from:
+            if not updated_at_name:
+                raise SQLMeshError(
+                    "Cannot use `updated_at_as_valid_from` without `updated_at_name` for SCD Type 2"
+                )
+            update_valid_from_start = updated_at_name
+        elif execution_time_as_valid_from:
+            update_valid_from_start = execution_ts
+        else:
+            update_valid_from_start = self._to_utc_timestamp(
+                "1970-01-01 00:00:00+00:00", time_data_type
+            )
+        insert_valid_from_start = execution_ts if check_columns else exp.to_column(updated_at_name)  # type: ignore
+        if check_columns:
+            row_check_conditions = []
+            for col in check_columns:
+                t_col = col.copy()
+                t_col.this.set("this", f"t_{col.name}")
```

Comment:
this is a bit tricky, could we do

t_col.set("this", exp.to_identifier(f"t_{col.name}") 

it's a bit more explicit imo, setting the idintefier this like you've done makes me feel antsy